packages/commonwealth/client/scripts/state/api/comments/editComment.ts

```diff
@@ -0,0 +1,89 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import axios from 'axios';
+import Comment from 'models/Comment';
+import app from 'state';
+import { ApiEndpoints } from 'state/api/config';
+import useFetchCommentsQuery from './fetchComments';
+
+interface EditCommentProps {
+  address: string;
+  chainId: string
+  parentCommentId: number | null,
+  threadId: number;
+  commentId: number;
+  updatedBody: string;
+  attachments?: string[];
+}
+
+const editComment = async ({
+  address,
+  chainId,
+  parentCommentId,
+  threadId,
+  commentId,
+  updatedBody,
+  attachments,
+}: EditCommentProps) => {
+  const {
+    session = null,
+    action = null,
+    hash = null,
+  } = await app.sessions.signComment({
+    thread_id: threadId,
+    body: updatedBody,
+    parent_comment_id: parentCommentId,
+  })
+
+  const response = await axios.patch(`${app.serverUrl()}/comments/${commentId}`, {
+    address: address,
+    author_chain: chainId,
+    id: commentId,
+    chain: chainId,
+    body: encodeURIComponent(updatedBody),
+    'attachments[]': attachments,
+    jwt: app.user.jwt,
+    canvas_action: action,
+    canvas_session: session,
+    canvas_hash: hash,
+  })
+
+  return new Comment(response.data.result)
+};
+
+interface UseEditCommentMutationProps {
+  chainId: string
+  threadId: number;
+}
+
+const useEditCommentMutation = ({ chainId, threadId }: UseEditCommentMutationProps) => {
+  const queryClient = useQueryClient();
+  const { data: comments } = useFetchCommentsQuery({
+    chainId,
+    threadId,
+  })
+
+  return useMutation({
+    mutationFn: editComment,
+    onSuccess: async (updatedComment) => {
+      // find the existing comment index
+      const foundCommentIndex = comments.findIndex(x => x.id === updatedComment.id)
```

Comment:
I would prefer if we could use `id` instead of `index`. Then below it simply would be 
```   
 return comments.map(comment => {
      if (comment.id === updatedComment.id) {
        return updatedComment
      }
      
      return comment
    })  
```

Reply:
I am not gonna fight here but imo in this case readability is a king, not a premature optimisation. We are talking here about maybe dozens of comments, not millions, so there won't be any visible difference.

Reply:
Updated (I went with the map)